presto-hive/src/main/java/com/facebook/presto/hive/HiveClientModule.java

```diff
@@ -77,14 +77,7 @@ public void configure(Binder binder)
         binder.bind(HiveSessionProperties.class).in(Scopes.SINGLETON);
         binder.bind(HiveTableProperties.class).in(Scopes.SINGLETON);
 
-        if (metastore != null) {
-            binder.bind(HiveMetastore.class).toInstance(metastore);
-        }
-        else {
-            binder.bind(HiveMetastore.class).to(CachingHiveMetastore.class).in(Scopes.SINGLETON);
-            newExporter(binder).export(HiveMetastore.class)
-                    .as(generatedNameOf(CachingHiveMetastore.class, connectorId));
-        }
+        bindMetastore(binder);
```

Comment:
Why is this change necessary? The method does the same thing and I don't see you overriding it. If you just need to swap the CachingHiveMetastore impl for another one, in this cases I'd say better to make the 2 line change to this class to do so.